### PR TITLE
Added new check HEADER_NOT_BEFORE_PACKAGE

### LINKED ISSUE #5: [Rule 2.4] Warn and fix if header comment is placed before package and imports directives
Can be detected as follows: warn if there are two consecutive KDocs - i.e. one is misplaced top-level, which isn't bound to any entity, and another one is KDoc for the next entity. Ensure the former is at the top of the file, right after the copyright block if it is necessary.

### PR DESCRIPTION
**Pull request by petertrr**
### What's done:
* Added new warnings with check and fix
* Added new tests

### TODO
* [x] Update README.md

This closes #5

### CODE REVIEW

File: diktat-rules/src/main/kotlin/org/cqfn/diktat/ruleset/rules/comments/HeaderCommentRule.kt [hunk 1]
```diff
@@ -80,25 +96,47 @@ class HeaderCommentRule : Rule("header-comment") {
...
+     * @return true if position check is not needed or if header KDoc is positioned correctly or it was moved by fix mode
+     */
+    private fun checkHeaderKdocPosition(node: ASTNode): Boolean {
+        // AST always has PACKAGE_DIRECTIVE and IMPORT_LIST nodes, even if they are empty
+        if (configRules.isRuleEnabled(HEADER_NOT_BEFORE_PACKAGE) && node.findChildBefore(PACKAGE_DIRECTIVE, KDOC) == null) {
+            val firstKdoc = node.findChildAfter(IMPORT_LIST, KDOC)
+            if (firstKdoc != null && firstKdoc.treeParent.elementType == FILE) {
+                HEADER_NOT_BEFORE_PACKAGE.warnAndFix(configRules, emitWarn, isFixMode, fileName, firstKdoc.startOffset) {
+                    node.moveChildBefore(firstKdoc, node.getFirstChildWithType(PACKAGE_DIRECTIVE), true)
+//                         remove empty line before copyright and header kdoc
```
[the remark this answers is not in the document]
Comment (petertrr) [reply diktat-rules/src/main/kotlin/org/cqfn/diktat/ruleset/rules/comments/HeaderCommentRule.kt]: You have commented on code from one of older commits, this line is removed in last version.

File: diktat-rules/src/main/kotlin/org/cqfn/diktat/ruleset/constants/Warnings.kt [hunk 2]
```diff
@@ -55,8 +55,9 @@ enum class Warnings(private val id: Int, private val canBeAutoCorrected: Boolean
     HEADER_MISSING_OR_WRONG_COPYRIGHT(37, true, "file header comment must include copyright information inside a block comment"),
     HEADER_CONTAINS_DATE_OR_AUTHOR(38, false, "file header comment should not contain creation date and author name"),
     HEADER_MISSING_IN_NON_SINGLE_CLASS_FILE(39, false, "files that contain multiple or no classes should contain description of what is inside of this file"),
-    COMMENTED_OUT_CODE(40, false, "you should not comment out code, use VCS to save it in history and delete this block"),
-    FILE_IS_TOO_LONG(41, false, "file has more number of lines than expected")
+    HEADER_NOT_BEFORE_PACKAGE(40, true, "header KDoc should be placed before package and imports"),
```
Comment (akuleshov7) [inline diktat-rules/src/main/kotlin/org/cqfn/diktat/ruleset/constants/Warnings.kt:58]: you will see conflicts right now )

File: diktat-rules/src/main/kotlin/org/cqfn/diktat/ruleset/rules/comments/HeaderCommentRule.kt [hunk 3]
```diff
@@ -82,6 +96,32 @@ class HeaderCommentRule : Rule("header-comment") {
...
+     * @return true if position check is not needed or if header KDoc is positioned correctly or it was moved by fix mode
+     */
+    private fun checkHeaderKdocPosition(node: ASTNode): Boolean {
+        // AST always has PACKAGE_DIRECTIVE and IMPORT_LIST nodes, even if they are empty
+        if (configRules.isRuleEnabled(HEADER_NOT_BEFORE_PACKAGE) && node.findChildBefore(PACKAGE_DIRECTIVE, KDOC) == null) {
+            val firstKdoc = node.findChildAfter(IMPORT_LIST, KDOC)
+            if (firstKdoc != null && firstKdoc.treeParent.elementType == FILE) {
+                HEADER_NOT_BEFORE_PACKAGE.warnAndFix(configRules, emitWarn, isFixMode, fileName, firstKdoc.startOffset) {
+                    node.moveChildBefore(firstKdoc, node.getFirstChildWithType(PACKAGE_DIRECTIVE), true)
+                    // ensure there is no empty line between copyright and header kdoc
```
Comment (akuleshov7) [inline diktat-rules/src/main/kotlin/org/cqfn/diktat/ruleset/rules/comments/HeaderCommentRule.kt:110]: static analysis will tell you that you have too many levels of nesting

File: diktat-rules/src/main/kotlin/org/cqfn/diktat/ruleset/constants/Warnings.kt (hunk 2 above)
Comment (petertrr) [reply diktat-rules/src/main/kotlin/org/cqfn/diktat/ruleset/constants/Warnings.kt:58]: Do you suggest to only append new warnings at the end of list? I try to keep them in order in which they appear in our doc, so I'm ready to resolve conflicts :)

File: diktat-rules/src/main/kotlin/org/cqfn/diktat/ruleset/rules/comments/HeaderCommentRule.kt (hunk 3 above)
Comment (petertrr) [reply diktat-rules/src/main/kotlin/org/cqfn/diktat/ruleset/rules/comments/HeaderCommentRule.kt:110]: I see your point and will fix, but the check has been successful though...

**Pull request merged**